src/main/java/org/broadinstitute/dsm/jobs/PubSubLookUp.java

```diff
@@ -0,0 +1,144 @@
+
+        //corrected result -> assuming corrected results are always changed
+        if (dsmTestResult != null && !testBostonResult.isCorrected() && dsmTestResult.isCorrected) {
+            return true;
+        }
+        //duplicate result
+        if (testBostonResult.isCorrected() == dsmTestResult.isCorrected
+                && StringUtils.isNotBlank(dsmTestResult.result) && dsmTestResult.result.equals(testBostonResult.getResult())
+                && StringUtils.isNotBlank(dsmTestResult.date) && dsmTestResult.date.equals(testBostonResult.getTimeCompleted())) {
+            return false;
+        }
+        // weird result
+        if (!testBostonResult.isCorrected()
+                && StringUtils.isNotBlank(dsmTestResult.result) && !dsmTestResult.result.equals(testBostonResult.getResult()) ) {
+
+            logger.error("A new result for sample id "+testBostonResult.getSampleId() +" that doesn't match the previous one. Date of the new result: "+testBostonResult.getTimeCompleted());
+            return false;
+        }
+        return true;
+    }
+
+    private static void tellPepperAboutTheNewResults(TestBostonResult testBostonResult) {
+        String query = "select   eve.event_name,   eve.event_type,   request.ddp_participant_id,   request.dsm_kit_request_id,   realm.ddp_instance_id,   realm.instance_name,   realm.base_url,   realm.auth0_token,   realm.notification_recipients,   realm.migrated_ddp,   kit.receive_date,   kit.scan_date   from   ddp_kit_request request,   ddp_kit kit,   event_type eve,   ddp_instance realm   where request.dsm_kit_request_id = kit.dsm_kit_request_id   and request.ddp_instance_id = realm.ddp_instance_id   " +
+                " and (eve.ddp_instance_id = request.ddp_instance_id   and eve.kit_type_id = request.kit_type_id)   and eve.event_type = \"RESULT\" " + // that's the change from the original query
+                " and request.external_order_number = ?"; // that's the change from the original query
+
+        KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(query, testBostonResult.getSampleId());
+        if (kitDDPNotification != null) {
+            EventUtil.triggerDDPWithTestResult(kitDDPNotification, testBostonResult);
+        }
+    }
+
+    private static DSMTestResult[] getLatestKitTestResults(TestBostonResult testBostonResult) {
+        String externalOrderNumber = testBostonResult.getSampleId();
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SELECT_LATEST_RESULT_QUERY)) {
+                stmt.setString(1, externalOrderNumber);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    if (rs.next()) {
+                        String resultArray = rs.getString("test_result");
+                        DSMTestResult[] dsmTestResultsArray = new Gson().fromJson(resultArray, DSMTestResult[].class);
```

Comment:
Hmmm interesting, so idea is to store it as a json array instead of one per database row?

Reply:
Yes because there is the possibility of having more than one per each kit, right? (considering revised and corrected ones )